src/tcp.c

```diff
@@ -145,6 +145,13 @@ enftun_tcp_connect(struct enftun_tcp* tcp,
             break;
     }
 
+    if (addr != NULL) {
+        tcp->remote_addr = *addr->ai_addr;
```

Comment:
This won't copy the entire address.  It will only copy the first `sizeof(struct sockaddr)` bytes. But the actual data has either `sizeof(struct sockaddr_in)` or `sizeof(struct sockaddr_in6)` bytes.

I suggest using `getpeername` to get the remote addr, similar to using `getsockname` to get the local address.